server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexStateService.java

```diff
@@ -176,27 +168,29 @@ public void onFailure(final String source, final Exception e) {
                                         public void clusterStateProcessed(final String source,
                                                                           final ClusterState oldState, final ClusterState newState) {
 
-                                            final String[] indices = results.entrySet().stream()
-                                                .filter(result -> result.getValue().isAcknowledged())
-                                                .map(result -> result.getKey().getName())
-                                                .filter(index -> newState.routingTable().hasIndex(index))
+                                            final List<IndexResult> indices = new ArrayList<>(results.values());
+                                            final boolean acknowledged = indices.stream().noneMatch(IndexResult::hasFailures);
```

Comment:
In `closeRoutingTable`, we check for:
```
currentState.blocks().hasIndexBlock(index.getName(), closingBlock) == false
```
I think this will not be matched anymore by this acknowledged check, meaning acknowledged would be true afterwards?

Also, I would think that if the block was gone and the index thus is left open, we should see a failure for it in the response?

Reply:
Good observation :). I pushed https://github.com/elastic/elasticsearch/pull/39687/commits/434696b11931b378863b47ed8b03760d5c9b12f2 to report error if closing block disappeared.